### PR TITLE
Terraform tutorial

### PR DESCRIPTION
**Pull request by erasche**
This mostly covers use of terraform of VGCN (this specifically uses a 'denbi' image, but that's just a VGCN-lite version.)

A VGCN specific one will follow which will depend on this tutorial as prerequisite knowledge for launching a VGCN cluster which will attach to Galaxy with pulsar.

- Adds slides for ansible
- Slides for Terraform
- Hands on for terraform

### CODE REVIEW

File: topics/admin/tutorials/terraform/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,992 @@
...
+contributors:
+  - erasche
+---
+
+# Overview
+{:.no_toc}
+
+In this tutorial we have briefly cover what Terraform is and how you can leverage it for your needs. This will not make you an expert on Terraform but will give you the tools you need in order to maintain your cloud infrastructure as code.
+
+This will be a very practical training with emphasis on looking at examples from modules and becoming self sufficient. This tutorial uses OpenStack.
```
Comment (bgruening) [inline topics/admin/tutorials/terraform/tutorial.md]: This tutorial uses the OpenStack adapter of Terraform.

File: topics/admin/tutorials/terraform/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,992 @@
...
+>    Apply complete! Resources: 1 added, 0 changed, 0 destroyed.
+>    ```
+{: .hands_on}
+
+We should now have a keypair in our cloud!
+
+
+## Adding an Instance
+
+We've now have:
```
Comment (bgruening) [inline topics/admin/tutorials/terraform/tutorial.md]: We now have?

File: topics/admin/tutorials/terraform/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,992 @@
...
+>    [centos@test-vm ~]$
+>    ```
+>
+{: .hands_on}
+
+With this, you're done. You've launched a single VM with Terraform. The image we are using comes with:
+
+- docker
+- singularity
+- conda
```
Comment (bgruening) [inline topics/admin/tutorials/terraform/tutorial.md]: condor as well?

**Comment by erasche** [comment]
@bgruening hopefully it is good for the de.NBI workshop on tuesday, and will extend with a VGCN specific one to make this more "galaxy" related.

File: topics/admin/tutorials/terraform/tutorial.md [hunk 4]
```diff
@@ -0,0 +1,1194 @@
...
+Note: You didn't specify an "-out" parameter to save this plan, so Terraform
+can't guarantee that exactly these actions will be performed if
+"terraform apply" is subsequently run.
+```
+
+Terraform informs us first about the different symbols used. Here it tells us that it will `+ create` a resource. Sometimes it will `- delete` or `~ update in-place`. Next it describes in detail what will be done and why. For creating a resource it does not give us much information, we will see more types of changes later.
+
+Lastly it informs us that we did not save our plan. Terraform can maintain a concept of what the remote resource's state looks like between your `terraform plan` and `terraform apply` steps. This is a more advanced feature and will not be covered today.
+
+> ### {% icon hands_on %} Applying our plan
```
Comment (shiltemann) [inline topics/admin/tutorials/terraform/tutorial.md]:
```suggestion
> ### {% icon hands_on %} Hands-on: Applying our plan
```

**Comment by bgruening** [comment]
Thanks a lot @erasche and @shiltemann!

**Pull request merged**